csharp/ql/src/semmle/code/csharp/Generics.qll

```diff
@@ -23,19 +23,19 @@ private import dotnet
  */
 class Generic extends DotNet::Generic, Declaration, @generic {
   Generic() {
-    is_generic(this) or
-    is_constructed(this)
+    type_parameters(_, _, this, _) or
```

Comment:
These lines are repeated in the derived QL classes. So perhaps make `class Generic` `abstract`?

Reply:
We tend to avoid `abstract` classes in the library, as it makes subclassing very hard.